### PR TITLE
make compatible with TYPO3 12

### PR DESCRIPTION
**Pull request by Kanti**
Tested in TYPO3 11 and 12

### CODE REVIEW

File: Configuration/Services.yaml [hunk 1]
```diff
@@ -37,3 +47,10 @@ services:
                 command: 'crontab:import-scheduler-tasks'
                 description: 'Import Scheduler Tasks.'
                 schedulable: false
+
+    Helhum\TYPO3\Crontab\EventListener\BeforeModuleCreationEventListener:
+        tags:
+            -
+                name: 'event.listener'
+                identifier: 'helhum/t-y-p-o3/crontab/before-module-creation-event-listener'
```
Comment (sascha-egerer) [inline Configuration/Services.yaml]:
```suggestion
                identifier: 'helhum/crontab/before-module-creation-event-listener'
```

`/t-y-p-o3/`?

Comment (Kanti) [reply Configuration/Services.yaml]: it was auto generated by b13/make

**Comment by linawolf** [comment]
@helhum what is the plan here?

**Comment by linawolf** [comment]
According to me tests this solution works with scheduler tasks but not with console commands as there will be a type error in the CommandExecutor. @Kanti have you tested your code in TYPO3 v12 with a console command?

**Comment by Kanti** [comment]
We are running this in Production for about 6 Months without Problems. (with Console Commands)
What is the error you are getting? with what PHP Version?

**Comment by linawolf** [comment]
In the CommandExecutor, line 37 an array is passed to ArgvInput which expects an InputInterface or null

**Comment by devtroll** [comment]
@linawolf @Kanti Merge Status to Main?

**Comment by Kanti** [comment]
I see this extension has 2 features.

-  It allowed to commit the defined scheduler tasks.
- and it tracks the processid and marked task as not running If they had a problem 

One of these features can possible done with events in the scheduler. (Processid)

The other one can be done with an import into the scheduler tasks table.

I think an extension that dose both of these things should be a good solution without much implementation. And using the existing infrastructure from the Core/Scheduler.

The feature that all task are executed in separate process could maybe brought into the core or be done with an xclass. I think that should not be a priority.

**Comment by devtroll** [comment]
@Kanti The Main Problem of TYPO3 Scheduler is the Sharing between multiple Instances. That is Not possible

**Comment by helhum** [comment]
I was not able to push additional changes on top of this, so I added those commits to #18 and will close this one now

**Comment by helhum** [comment]
Thanks a ton @Kanti for the preparation! Thanks everybody for your patience!

**Pull request closed without merging**